File: BendApp1/Models/Thing.cs
namespace BendApp1.Models;
public class Thing
{
    public int id { get; set; }
    public int areaId { get; set; }
    public int? joinedWith { get; set; }
    public string sku { get; set; }
    public string defaultSku { get; set; }
    public string status { get; set; }
    public string countActive { get; set; }
}

File: BendApp1/ClientApp/src/app/core/models/thing/thing.ts
export class Thing {
    id!: number;
    areaId!: number;
    joinedWith!: number;
    sku!: string;
    defaultSku!: string;
    status!: string;
    countActive!: number;

    getStatusClass(group: Thing[]): string {
        const sumCountActive = group.reduce((sum, thing) => sum + thing.countActive, 0);
        if (this.status === 'closed') {
            return 'status status-closed';
        } else if (sumCountActive !== 0) {
            return 'status status-active';
        } else {
            return 'status status-opened';
        }
    }
}
